File: 1567b_mm.cpp
/**
 * @file 1567b_mm.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem MEXor
 * Mixup(https://codeforces.com/problemset/problem/1567/B)
 * @version 0.1
 * @date 2024-08-31
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int t, a, b, res, i;
    map<int, int> pcxor;

    for (i = 1, res = 0; i <= 300000; i++) {
        pcxor[i] = res;
        res ^= i;
    }

    cin >> t;
    while (t--) {
        cin >> a >> b;
        if (pcxor[a] == b)
            cout << a;
        else
            (pcxor[a] ^ b) != a ? cout << a + 1 : cout << a + 2;
        cout << '\n';
    }
    return (EXIT_SUCCESS);
}

File: 1567b_mm.py
"""
@file: 1567b_mm.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem MEXor Mixup(https://codeforces.com/contest/1567/problem/B)
@version: 1.0
@date: 2025-02-03

@copyright: Copyright (c) 2025
"""

from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(string):
    return stdout.write(str(string) + "\n")


def main():
    for _ in range(int(input())):
        a, b = [int(x) for x in input().split()]
        res = [i for i in range(a)]
        cxor = 0
        for i in res:
            cxor ^= i
        print(a if cxor == b else a + 1 if cxor ^ b != a else a + 2)


if __name__ == "__main__":
    main()
